File: python/files/dir_walk.py
#!/usr/bin/env python

import os
import sys

dir_input = sys.argv[1]


def get_files(dir: str):
    for root, dir_names, file_names in os.walk(dir):
        for file in file_names:
            yield os.path.join(root, file)


for file in get_files(dir_input):
    print(file)

File: clojure/files/dir_walk.clj
#!/usr/bin/env bb
; uses babashka clojure to run
; brew install borkdude/brew/babashka
; see https://github.com/borkdude/babashka
; KEYS=[dir file_system walk iterate file]

(defn get-files [dir]
  (file-seq (clojure.java.io/file "/tmp")))


(defn print-files [dir]
  (doseq [file (get-files dir)]
    (prn file)))

(let [[dir] *command-line-args*]
  (when (empty? dir)
     (println "Usage: <directory>")
     (System/exit -1))
  (print-files dir))
